inject-java/src/test/groovy/io/micronaut/inject/provider/BeanProviderSpec.groovy

```diff
@@ -231,30 +231,91 @@ class BeanOneTwo implements BeanNumber { }
 @jakarta.inject.Singleton
 class Test {
     public BeanProvider<BeanNumber> provider;
-    Test(@OneQualifier BeanProvider<BeanNumber> provider) {
+    public BeanProvider<BeanNumber> providerTwo;
+    Test(@OneQualifier BeanProvider<BeanNumber> provider, @TwoQualifier BeanProvider<BeanNumber> providerTwo) {
         this.provider = provider;
+        this.providerTwo = providerTwo;
     }
 }
 ''')
-        when:
+        when: 'retrieve test bean'
         def bean = getBean(context, 'test.Test')
 
-        then:
-        bean.provider.isPresent()
```

Comment:
Why did you change this? Preferably do not modify existing checks without explanation

Reply:
This is the test for #7133. So, I've improved `get` method test a little bit, and added the same test for `find` method. Changes in this PR touches both `get` and `find` code paths.

I can move changes to `get` method tests to separate commit if you prefer, or rollback them completely.

Reply:
Just return that `bean.provider.isPresent()` check

Reply:
Ok, may I force force push? Or I should do a separate commit?

Reply:
You can force push

Reply:
Done. Also added the same check to `get` method test.